packages/airnode-node/src/workers/local-gateways/validation.ts

```diff
@@ -102,6 +120,170 @@ export function verifyHttpSignedDataRequest(
   return { success: true, encodedParameters, endpointId: validEndpointId };
 }
 
+const beacon = z.object({
+  airnodeAddress: z.string(),
+  endpointId: z.string(),
+  encodedParameters: z.string(),
+  encodedValue: z.string().optional(),
+  timestamp: z.string().optional(),
+  signature: z.string().optional(),
+});
+
+export const signOevDataBodySchema = z.object({
+  chainId: z.number().positive(),
+  dapiServerAddress: z.string(),
+  oevProxyAddress: z.string(),
+  updateId: z.string(),
+  bidderAddress: z.string(),
+  bidAmount: z.string(),
+  signedData: z.array(beacon),
+});
+
+type Beacon = z.infer<typeof beacon>;
+interface BeaconDecoded extends Required<Beacon> {
+  decodedValue: ethers.BigNumber;
+}
+
+export type ProcessSignOevDataRequestBody = z.infer<typeof signOevDataBodySchema>;
+
+export function dropInvalidBeacons(beacons: Beacon[]) {
+  const currentTime = new Date();
+
+  return beacons.reduce((result, beacon) => {
+    const goProcessing = goSync(() => {
+      const { airnodeAddress, endpointId, encodedParameters, encodedValue, timestamp, signature } = beacon;
+
+      // We don't have data for this beacon
+      if (!encodedValue || !timestamp || !signature) {
+        throw new Error('Missing beacon data');
+      }
+
+      // The timestamp is older than the last 2 minutes
+      if (isBefore(toDate(Number(timestamp)), addMinutes(currentTime, -TIMESTAMP_DEVIATION))) {
+        throw new Error('Beacon timestamp too old');
+      }
+
+      // To check parameters validity, exception is caught by the goSync
+      const _decodedParameters = decode(encodedParameters);
+      const decodedValue = ethers.BigNumber.from(ethers.utils.defaultAbiCoder.decode(['int256'], encodedValue)[0]);
+
+      const template: ApiCallTemplateWithoutId = {
+        airnodeAddress,
+        endpointId,
```

Comment:
You are missing endpointId check. See `verifyEndpointId` function and update it for OEV as well.

Reply:
But there's not much I can verify about endpoint ID. In the `verifyEndpointId` function the main check is whether the endpoint ID exists among triggers which we don't have here. And the other check is just via Zod schema. But that's just checking whether it's a string and that's already done in the request body check.

Reply:
But we can check that it is one of the endpoints, since all endpoints are defined in config.json.

Reply:
That's not true. We're sending data from all the Airnodes. Yes, some would pass but you won't be (most likely) able to find all of them. The question is whether it's even worth it in this case.

Reply:
The ones the Airnode is signing must be from it's config.json.

Reply:
Yeah, I know, that's what I'm saying. So, I can check that those are present.

Reply:
Hmm, but I can't check that. The endpoint ID is never used anywhere else but the triggers. It is usually derived from `oisTitle` and `endpointName` but that's just a convention so I can't even derive it to compare it. I would skip this tbh. Is there any harm in that?

Reply:
I guess not... You won't be able to have valid signed data for this anyway (since we sign with template ID and endpoint ID is part of that).